R/PackageInheritanceReporter.R

```diff
@@ -0,0 +1,212 @@
+#' @title Package Class Inheritance Reporter Class
+#' @name InheritanceReporter
+#' @family PackageReporters
+#' @description This reporter takes a package and traces the class inheritance
+#'   structure. Currently the following object-oriented systems are supported:
+#'   \itemize{
+#'       \item{S4 Classes}
+#'       \item{Reference Classes (sometimes informally called "R5")}
+#'       \item{R6 Classes}
+#'   }
+#'   S3 classes are not supported, as their inheritance is defined on an ad hoc
+#'   basis per object and not formally by class definitions.
```

Comment:
S3 doesn't have "classes", right? Doesn't it just give you a framework for generic dispatching on existing types?

Reply:
No, you can define arbitrary new classes ad hoc and write new functions that get those classes will dispatch to. 

https://adv-r.hadley.nz/s3.html#s3-classes